arangod/Network/ConnectionPool.cpp

```diff
@@ -199,11 +189,14 @@ std::shared_ptr<fuerte::Connection> ConnectionPool::createConnection(fuerte::Con
 
 ConnectionPtr ConnectionPool::selectConnection(std::string const& endpoint,
                                                ConnectionPool::Bucket& bucket) {
+  const std::chrono::milliseconds ttl(_config.idleConnectionMilli);
+  auto now = std::chrono::steady_clock::now();
+  
   std::lock_guard<std::mutex> guard(bucket.mutex);
 
   for (std::shared_ptr<Context>& c : bucket.list) {
-    const fuerte::Connection::State state = c->fuerte->state();
-    if (state == fuerte::Connection::State::Closed) {
+    if (c->fuerte->state() == fuerte::Connection::State::Closed ||
+        (now - c->lastLeased) > ttl) {
```

Comment:
I understand that we cannot select a connection that was already closed. But why can't we select a connection anymore that has reached its expiration time? 
It should be ok to select an expired connection here and use it if the conection is not yet closed.
Pruning expired connections and selecting connections both acquire the bucket's mutex, so the two operations should never run at the same time on the same connection.

Reply:
no expired connections are _expired_

Reply:
Fine, but are expired connections non-functional anymore?
My take is that an expired connection could still be reused/selected if otherwise ok. So we can save creating/opening a new connection object in the best case.
I am happy if there is a reason why we can't reuse expired connections, it's just that I currently don't understand it completely. Can you clarify?

Reply:
Read the title:
As I said in the team meeting: "the server side closes unused connections after the specified keep alive timeout. If the client doesn't do the same you get an error when trying to use the connection"